samples/AzureOpenAI/ReadMe.md

```diff
@@ -1,19 +1,57 @@
 
-Before publishing the extension, make sure to initialize the secrets in the codeunit `SecretsAndCapabilitiesSetup.Codeunit.al` and the app ID in `app.json`.
+> **_NOTICE:_** These apps have been tested with the Azure OpenAI models that were newest when the sample was created. If you install them today they might use newer models that require tweaking of the prompts to work properly.
 
-The completions endpoint typically has the format of: `https://<resource name>.openai.azure.com/openai/deployments/<deployment name>/completions?api-version=2022-12-01`.
-This sample has been tested with the `gpt-35-turbo-16k` model. Using another model may require tweaking of the prompts (for example with turbo or gpt4).
 
 ## Calling Azure OpenAI
-
-You can find a minimal example on how to call the Chat Completion endpoint inside `GenerateItemSubProposal.Codeunit.al`, procedure `Chat`.
-
 In general, large language models sometimes provide information that is not accurate. Make sure you implement all the necessary steps to ensure you avoid these cases, including but not limited to:
-- Check grounding for the facts that Azure OpenAI returns (in this example, checking that the suggested item exists is our grounding step)
+- Check grounding for the facts that Azure OpenAI returns (for example, checking that the suggested item exists is our grounding step, or make sure that the job/project suggestion has estimated completion dates in the future)
 - Always make the user review any suggestion from Copilot before you save the changes
 - When implementing your own scenarios, ensure you follow Responsible AI design patterns. For more information, check out https://aka.ms/RAI
+
+
+# Folders
+### 1-GetStarted
+This folder contains the simplest example of an extension that adds a new Copilot feature/capability to Business Central.
+
+This extension is ready for you to explore. If you are NOT using Managed AI Resources, you will get an error you get an error similar to "The authorization is not configured". You need to change the line in `DraftProject.Page.al`, and instead of `AzureOpenAI.SetManagedResourceAuthorization` call `AzureOpenAI.SetAuthorization` specifying your own Azure OpenAI endpoint, deployment, and API key. The endpoint typically has the format of: `https://<resource name>.openai.azure.com/`.
+
+After you install the extension, you will be able to access its logic from the existing Base Application page `page 89 "Job List"`.
+
+
+### 2-ItemSubstitution
+This folder contains one of the early samples that uses the AL Copilot Toolkit. It demonstrates how to use the PromptDialog page type and its special capabilities, as well as how to set up and tweak the parameters to call Azure Open AI from AL.
+
+This extension is ready for you to explore. Before you install it, you need to modify the file `SecretsAndCapabilitiesSetup.Codeunit.al`, and make sure Isolated Storage is set up with your Azure OpenAI endpoint, deployment, and API key.
```

Comment:
Aren't we switching everywhere for Managed resources?

Reply:
Yes, but that still requires endpoint, deployment, and API key. At least for the moment being.